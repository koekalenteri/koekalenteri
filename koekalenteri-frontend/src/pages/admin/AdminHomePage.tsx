import { useAuthenticator } from '@aws-amplify/ui-react';
import { Box, Toolbar } from '@mui/material';
import { useState } from 'react';
import { Navigate, Outlet, useLocation } from 'react-router-dom';
import { Header, SideMenu } from '../../layout';

export function AdminHomePage() {
  const location = useLocation();
  const { route } = useAuthenticator(context => [context.route]);
  const [menuOpen, setMenuOpen] = useState(false);

  return (route !== 'authenticated' ? <Navigate to="/login" state={{ from: location }} replace /> :
    <>
      <Header title={'Admin'} toggleMenu={() => setMenuOpen(!menuOpen)} />
      <Box sx={{ display: 'flex', height: '100%' }}>
        <SideMenu open={menuOpen} onClose={() => setMenuOpen(false)} />
        <Box sx={{ p: 1, display: 'flex', flexDirection: 'column', flexGrow: 1, overflow: 'auto' }}>
          <Toolbar variant="dense" />
          <Outlet />
        </Box>
      </Box>
    </>
  )
}
