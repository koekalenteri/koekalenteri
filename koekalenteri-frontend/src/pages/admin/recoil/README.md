# State available to authenticated users

These flowcharts describes the relationship between atoms, selectors, effects and persistent storage.

## Events

```mermaid
graph LR
  adminEventsAtom-->remoteAdminEventsEffect
  adminEventsAtom-->storageEffect
  newEventAtom-->storageEffect
  adminShowPastEventsAtom-->storageEffect
  adminEventFilterTextAtom-->storageEffect
  adminEventIdAtom-->storageEffect
  eventClassAtom-->storageEffect
  editableEventByIdAtom-->storageEffect

  storageEffect[(localStorage)]
  remoteAdminEventsEffect-->getEvents[/getEvents/]-->aws

  adminEventSelector-->newEventAtom
  adminEventSelector-->adminEventsAtom

  currentAdminEventSelector-->editableEventByIdAtom

  filteredAdminEventsSelector-->adminEventsAtom
  filteredAdminEventsSelector-->adminEventFilterTextAtom
  filteredAdminEventsSelector-->adminShowPastEventsAtom

  aws[(cloud)]
```

## Officials

```mermaid
graph LR
  officialsAtom-->storageEffect
  officialsAtom-->remoteOfficialsEffect
  officialFilterAtom

  storageEffect[(localStorage)]
  remoteOfficialsEffect-->getOfficials[/getOfficials/]-->aws[(cloud)]

  filteredOfficialsSelector-->officialFilterAtom
  filteredOfficialsSelector-->officialsAtom
```

## Organizers

```mermaid
graph LR
  organizersAtom-->storageEffect
  organizersAtom-->remoteOrganizersEffect
  organizersFilterAtom

  storageEffect[(localStorage)]
  remoteOrganizersEffect-->getOrganizers[/getOrganizers/]-->aws[(cloud)]

  filteredOrganizersSelector-->organizersFilterAtom
  filteredOrganizersSelector-->organizersAtom
```

## Registrations

```mermaid
graph LR
  adminRegistrationIdAtom-->storageEffect
  eventRegistrationsAtom-->storageEffect

  storageEffect[(localStorage)]

  currentEventRegistrationsSelector-->adminEventIdAtom
  currentEventRegistrationsSelector-->eventRegistrationsAtom

  currentEventClassRegistrationsSelector-->eventClassAtom
  currentEventClassRegistrationsSelector-->currentEventRegistrationsSelector

  currentAdminRegistrationSelector-->adminRegistrationIdAtom
  currentAdminRegistrationSelector-->currentEventClassRegistrationsSelector
```
