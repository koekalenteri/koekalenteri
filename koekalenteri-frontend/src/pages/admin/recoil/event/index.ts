export * from "./currentEvent"
export * from "./eventClassAtom"
export * from "./eventSelector"
