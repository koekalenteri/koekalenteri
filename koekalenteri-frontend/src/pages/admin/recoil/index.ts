export * from './emailTemplates'
export * from './events'
export * from './officials'
export * from './organizers'
export * from './registrations'
