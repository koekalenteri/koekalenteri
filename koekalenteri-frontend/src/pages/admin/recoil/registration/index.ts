export * from './currentEventClassRegistrationsQuery'
export * from './currentEventRegistrationsQuery'

export * from './registrationIdAtom'
export * from './registrationQuery'

export * from './registrationsQuery'
