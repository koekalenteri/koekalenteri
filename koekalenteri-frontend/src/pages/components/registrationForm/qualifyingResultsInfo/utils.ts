import type { ManualTestResult, QualifyingResult } from 'koekalenteri-shared/model'
import type { EventResultRequirement, EventResultRequirements, EventResultRequirementsByDate } from '../../../../rules'

import { nanoid } from 'nanoid'

import { unique } from '../../../../utils'
import { objectContains } from '../validation'

const asArray = (v: EventResultRequirements | EventResultRequirement) => (Array.isArray(v) ? v : [v])

function findFirstMissing(requirements: EventResultRequirementsByDate | undefined, results: QualifyingResult[]) {
  if (!requirements) {
    return []
  }
  for (const rule of requirements.rules) {
    for (const opt of asArray(rule)) {
      const { count, ...rest } = opt
      if (results.filter((r) => objectContains(r, rest)).length < count) {
        return rest
      }
    }
  }
}

export function availableTypes(requirements?: EventResultRequirementsByDate) {
  if (!requirements) {
    return []
  }
  return unique(requirements.rules.flatMap((rule) => asArray(rule).map((opt) => opt.type)))
}

export function availableResults(requirements?: EventResultRequirementsByDate, type?: string) {
  if (!requirements) {
    return []
  }
  return unique(
    requirements.rules.flatMap((rule) =>
      asArray(rule)
        .filter((opt) => !type || opt.type === type)
        .map((opt) => (opt.cert ? 'CERT' : opt.result))
    )
  )
}

export function createMissingResult(
  requirements: EventResultRequirementsByDate | undefined,
  results: ManualTestResult[],
  regNo: string
): ManualTestResult {
  const rule = findFirstMissing(requirements, results)
  return {
    id: nanoid(10),
    regNo,
    date: new Date(),
    official: false,
    qualifying: true,
    type: '',
    judge: '',
    location: '',
    result: '',
    class: '',
    ...rule,
  }
}

export function resultBorderColor(qualifying: boolean | undefined) {
  if (qualifying === true) {
    return 'success.light'
  }
  if (qualifying === false) {
    return 'error.main'
  }
}

export function getResultId(result: ManualTestResult | QualifyingResult) {
  if ('id' in result) {
    return result.id
  }
  return result.type + result.class + result.date.toISOString()
}
