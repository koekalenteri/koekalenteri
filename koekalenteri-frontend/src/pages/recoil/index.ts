export * from './effects'
export * from './events'
export * from './eventTypes'
export * from './judges'
export * from './officials'
export * from './organizers'
export * from './registration'
export * from './user'
