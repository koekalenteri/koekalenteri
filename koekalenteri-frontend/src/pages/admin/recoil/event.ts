import { EventEx } from "koekalenteri-shared/model"
import { atom, DefaultValue, selector, selectorFamily } from "recoil"

import { getEvent, getEvents } from "../../../api/event"
import { unique, uniqueDate } from "../../../utils"
import { logEffect, storageEffect } from "../../recoil/effects"

export interface DecoratedEvent extends EventEx {
  uniqueClasses: string[]
  uniqueClassDates: (eventClass: string) => Date[]
}

export const adminEventIdAtom = atom<string | undefined>({
  key: 'adminEventId',
  default: undefined,
  effects: [
    logEffect,
    storageEffect,
  ],
})

export const currentAdminEvent = selector({
  key: 'currentAdminEvent',
  get: ({ get }) => {
    const eventId = get(adminEventIdAtom)
    return eventId ? get(adminEventSelector(eventId)) : undefined
  },
  set: ({ set }, newValue) => {
    if (!newValue || newValue instanceof DefaultValue) {
      return
    }
    set(adminEventSelector(newValue.id), newValue)
  },
})

function decorateEvent(event: EventEx): DecoratedEvent {
  return {
    ...event,
    uniqueClasses: unique(event.classes.map(c => c.class)),
    uniqueClassDates: (eventClass: string) => uniqueDate(
      event.classes
        .filter(c => c.class === eventClass)
        .map(c => c.date || event.startDate || new Date()),
    ),
  }
}

export const adminEventsAtom = atom<DecoratedEvent[]>({
  key: 'adminEvents',
  default: getEvents().then(events => events.map(decorateEvent)),
  effects: [
    logEffect,
    storageEffect,
  ],
})

export const adminEventByIdAtom = selectorFamily<DecoratedEvent | undefined, string>({
  key: 'adminEvents/eventId',
  get: (eventId) => async ({ get }) => {
    let event = get(adminEventsAtom).find(event => event.id === eventId)
    if (!event) {
      const fetched = await getEvent(eventId)
      if (fetched) {
        event = decorateEvent(fetched)
      }
    }
    return event
  },
  set: (eventId) => ({ set }, newValue) => {
    if (!newValue || newValue instanceof DefaultValue) {
      return
    }
    set(adminEventsAtom, oldEvents => {
      const index = oldEvents.findIndex(item => item.id === eventId)
      const newEvents = oldEvents.map(event => ({...event}))
      newEvents.splice(index, 1, newValue)
      return newEvents
    })
  },
})

export const adminEventSelector = selectorFamily<DecoratedEvent | undefined, string>({
  key: 'adminEvent',
  get: (eventId) => ({get}) => get(adminEventByIdAtom(eventId)),
  set: (eventId) => ({set}, newValue) => set(adminEventByIdAtom(eventId), newValue),
})

export const eventClassAtom = atom<string | undefined>({
  key: 'eventClass',
  default: selector({
    key: 'eventClass/default',
    get: ({ get }) => get(currentAdminEvent)?.uniqueClasses?.[0],
  }),
  effects: [
    logEffect,
    storageEffect,
  ],
})
