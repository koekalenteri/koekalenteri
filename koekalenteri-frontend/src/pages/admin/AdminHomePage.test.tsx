import { Suspense } from 'react'
import { Authenticator } from '@aws-amplify/ui-react'
import { ThemeProvider } from '@mui/material'
import { render } from '@testing-library/react'
import { SnackbarProvider } from 'notistack'
import { RecoilRoot } from 'recoil'

import theme from '../../assets/Theme'
import { Path } from '../../routeConfig'
import { DataMemoryRouter, flushPromisesAndTimers } from '../../test-utils/utils'

import AdminHomePage from './AdminHomePage'

jest.useFakeTimers()

describe('AdminHomePage', () => {
  it('renders the page when user is logged in', async () => {
    const routes = [
      {
        path: Path.admin.root,
        element: <AdminHomePage />,
      },
      {
        path: Path.login,
        element: <>Login</>,
      },
    ]
    const { container } = render(
      <ThemeProvider theme={theme}>
        <RecoilRoot>
          <SnackbarProvider>
            <Authenticator.Provider>
              <Suspense fallback={<div>loading...</div>}>
                <DataMemoryRouter initialEntries={[Path.admin.root]} routes={routes} />
              </Suspense>
            </Authenticator.Provider>
          </SnackbarProvider>
        </RecoilRoot>
      </ThemeProvider>
    )

    await flushPromisesAndTimers()
    expect(container).toMatchSnapshot()
  })

  it('renders the child page content when user is logged in', async () => {
    const routes = [
      {
        path: Path.admin.root,
        element: <AdminHomePage />,
        children: [
          {
            path: Path.admin.index,
            element: <>ADMIN DEFAULT PAGE CONTENT</>,
          },
        ],
      },
      {
        path: Path.login,
        element: <>Login</>,
      },
    ]
    const { container } = render(
      <ThemeProvider theme={theme}>
        <RecoilRoot>
          <SnackbarProvider>
            <Authenticator.Provider>
              <Suspense fallback={<div>loading...</div>}>
                <DataMemoryRouter initialEntries={[Path.admin.index]} routes={routes} />
              </Suspense>
            </Authenticator.Provider>
          </SnackbarProvider>
        </RecoilRoot>
      </ThemeProvider>
    )

    await flushPromisesAndTimers()
    expect(container).toMatchSnapshot()
  })
})
