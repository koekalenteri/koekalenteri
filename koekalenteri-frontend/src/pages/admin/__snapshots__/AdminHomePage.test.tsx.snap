// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminHomePage renders the child page content when user is logged in 1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorSecondary MuiAppBar-positionFixed mui-fixed css-19iqa5y-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-dense css-izwrv4-MuiToolbar-root"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1ecacw7-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1flfxuu-MuiTypography-root-MuiLink-root"
        href="/"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-oau294-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <img
            alt="Suomen noutajakoirajärjestö"
            height="100%"
            src="snj-logo.png"
          />
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </a>
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-ix55se-MuiTypography-root-MuiLink-root"
        href="/"
      >
        <div
          class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1xz05ty-MuiTypography-root"
        >
          Koekalenteri
        </div>
      </a>
      <div
        class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1mo43ab-MuiTypography-root"
      >
         › Admin
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-kva05m-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="LanguageIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm6.93 6h-2.95c-.32-1.25-.78-2.45-1.38-3.56 1.84.63 3.37 1.91 4.33 3.56zM12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96zM4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2 0 .68.06 1.34.14 2H4.26zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56-1.84-.63-3.37-1.9-4.33-3.56zm2.95-8H5.08c.96-1.66 2.49-2.93 4.33-3.56C8.81 5.55 8.35 6.75 8.03 8zM12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96zM14.34 14H9.66c-.09-.66-.16-1.32-.16-2 0-.68.07-1.35.16-2h4.68c.09.65.16 1.32.16 2 0 .68-.07 1.34-.16 2zm.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95c-.96 1.65-2.49 2.93-4.33 3.56zM16.36 14c.08-.66.14-1.32.14-2 0-.68-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2h-3.38z"
            />
          </svg>
        </span>
        <div
          class="MuiBox-root css-130f8nx"
        >
          Suomeksi
        </div>
        <span
          class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ExpandMoreIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-kva05m-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="PersonOutlineIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 5.9c1.16 0 2.1.94 2.1 2.1s-.94 2.1-2.1 2.1S9.9 9.16 9.9 8s.94-2.1 2.1-2.1m0 9c2.97 0 6.1 1.46 6.1 2.1v1.1H5.9V17c0-.64 3.13-2.1 6.1-2.1M12 4C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 9c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </span>
        <div
          class="MuiBox-root css-130f8nx"
        >
          Test User
        </div>
        <span
          class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ExpandMoreIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </header>
  <div
    class="MuiBox-root css-ab8yd1"
  >
    <div
      class="MuiBox-root css-17swpt3"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-dense css-10ltm04-MuiToolbar-root"
      />
      ADMIN DEFAULT PAGE CONTENT
    </div>
  </div>
</div>
`;

exports[`AdminHomePage renders the page when user is logged in 1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorSecondary MuiAppBar-positionFixed mui-fixed css-19iqa5y-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-dense css-izwrv4-MuiToolbar-root"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1ecacw7-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1flfxuu-MuiTypography-root-MuiLink-root"
        href="/"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-oau294-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <img
            alt="Suomen noutajakoirajärjestö"
            height="100%"
            src="snj-logo.png"
          />
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </a>
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-ix55se-MuiTypography-root-MuiLink-root"
        href="/"
      >
        <div
          class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1xz05ty-MuiTypography-root"
        >
          Koekalenteri
        </div>
      </a>
      <div
        class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1mo43ab-MuiTypography-root"
      >
         › Admin
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-kva05m-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="LanguageIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm6.93 6h-2.95c-.32-1.25-.78-2.45-1.38-3.56 1.84.63 3.37 1.91 4.33 3.56zM12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96zM4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2 0 .68.06 1.34.14 2H4.26zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56-1.84-.63-3.37-1.9-4.33-3.56zm2.95-8H5.08c.96-1.66 2.49-2.93 4.33-3.56C8.81 5.55 8.35 6.75 8.03 8zM12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96zM14.34 14H9.66c-.09-.66-.16-1.32-.16-2 0-.68.07-1.35.16-2h4.68c.09.65.16 1.32.16 2 0 .68-.07 1.34-.16 2zm.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95c-.96 1.65-2.49 2.93-4.33 3.56zM16.36 14c.08-.66.14-1.32.14-2 0-.68-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2h-3.38z"
            />
          </svg>
        </span>
        <div
          class="MuiBox-root css-130f8nx"
        >
          Suomeksi
        </div>
        <span
          class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ExpandMoreIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-kva05m-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="PersonOutlineIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 5.9c1.16 0 2.1.94 2.1 2.1s-.94 2.1-2.1 2.1S9.9 9.16 9.9 8s.94-2.1 2.1-2.1m0 9c2.97 0 6.1 1.46 6.1 2.1v1.1H5.9V17c0-.64 3.13-2.1 6.1-2.1M12 4C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 9c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </span>
        <div
          class="MuiBox-root css-130f8nx"
        >
          Test User
        </div>
        <span
          class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ExpandMoreIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </header>
  <div
    class="MuiBox-root css-ab8yd1"
  >
    <div
      class="MuiBox-root css-17swpt3"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-dense css-10ltm04-MuiToolbar-root"
      />
    </div>
  </div>
</div>
`;
